src/util.h

```diff
@@ -418,13 +418,22 @@ class BufferValue : public MaybeStackBuffer<char> {
   } while (0)
 
 #define SPREAD_BUFFER_ARG(val, name)                                          \
-  CHECK((val)->IsArrayBufferView());                                          \
-  v8::Local<v8::ArrayBufferView> name = (val).As<v8::ArrayBufferView>();      \
-  v8::ArrayBuffer::Contents name##_c = name->Buffer()->GetContents();         \
-  const size_t name##_offset = name->ByteOffset();                            \
-  const size_t name##_length = name->ByteLength();                            \
-  char* const name##_data =                                                   \
-      static_cast<char*>(name##_c.Data()) + name##_offset;                    \
+  size_t name##_length;                                                       \
+  char* name##_data;                                                          \
+  if ((val)->IsArrayBuffer()){                                                \
+    v8::Local<v8::ArrayBuffer> name = (val).As<v8::ArrayBuffer>();            \
+    v8::ArrayBuffer::Contents name##_c = name->GetContents();                 \
+    name##_length = name##_c.ByteLength();                                    \
+    name##_data =                                                             \
+        static_cast<char*>(name##_c.Data());                                  \
```

Comment:
These two lines can be joined into one. Again, it would be great if you could fix that, but if not we can fix that while landing.